src/Dialect/ONNX/AdditionalONNXOps.td

```diff
@@ -140,16 +140,51 @@ def ONNXCustomOp:ONNX_Op<"Custom",
      DeclareOpInterfaceMethods<ShapeHelperOpInterface>]> {
   let summary = "ONNX Custom operation";
   let description = [{
+    This operation is not part of the standard and was added to support call
+    to any user defined external function.
     Allow call-out to a user defined operation. A single attribute
-    is a string which names the operation, other inputs are
-    passed to the user operation.
+    is a string which names the operation, other inputs are passed to the user
+    operation.
     The number of inputs and outputs can vary.
-
-    This operation is not part of the standard and was added to assist onnx-mlir.
+    Since the semantics of this op is not defined to compiler, some attributes
+    are introduced to guide shape inference:
+      'output_type':
+           Optional. The element type for the output tensor. Optional. If not
+           specified, follow the shape infer.
+      'shape_infer_pattern':
+           Optional. Specify how to propagate the shape info from inputs to
+           output. Supported patterns are `SameAs`, `MDBroadcast`.
+      'inputs_for_infer':
+           which inputs are used for shape inference. When mulitple
+           inputs are specified for 'SameAs', all these inputs should
+           have the same shape, and element type if output_type is
+           not specified. When this optional attribute is not specified,
+           all the inputs will be used for shape inference.
```

Comment:
`inputs_for_infer`: what is the format?

Reply:
this option does not seems to be used in shape inference.

Reply:
This option is intended to specify which inputs are used in shape inference, and is used to construct operands for ShapeHelper. For example, the broadcast pattern may be just on the first two input tensors. Then inputs_for_infer = [0, 1], and the other inputs will not be used in shape inference.

Reply:
please respond to the comment above